flatdata-rs/tests/coappearances/src/lib.rs

```diff
@@ -70,14 +70,11 @@ fn read_and_validate_coappearances() {
     );
 
     let validate_chapters = |edge_ref, expected| {
-        let e = edges.at(edge_ref);
-        let chapters_start = e.first_chapter_ref() as usize;
-        let chapters_end = edges.at(edge_ref + 1).first_chapter_ref() as usize;
+        let chapters_range = edges.at(edge_ref).chapters_range();
         let e_chapters: Vec<String> = g
             .chapters()
+            .slice(chapters_range.start as usize..chapters_range.end as usize)
```

Comment:
Annoying that there is no map.

Reply:
Yes, quite dislike it. How about adding another slice function that does casting for you?
```.slice_as_usize(chapters_range)```

Reply:
#135 created to track discussion